hack/dockerfiles/vendor.Dockerfile

```diff
@@ -38,7 +38,9 @@ RUN --mount=target=/context \
   fi
 EOT
 
-FROM psampaz/go-mod-outdated:${MODOUTDATED_VERSION} AS go-mod-outdated
+FROM --platform=linux/amd64 psampaz/go-mod-outdated:${MODOUTDATED_VERSION} AS go-mod-outdated-amd64
+
+FROM go-mod-outdated-amd64 AS go-mod-outdated
```

Comment:
Didn't catch this was not a multiplatform image, we should be able to build the tool in this Dockerfile instead.

Reply:
https://github.com/moby/buildkit/issues/5152